File: go/example_code/workdocs/wd_list_user_docs.go
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
// snippet-start:[workdocs.go.list_user_docs.complete]
package main

// snippet-start:[workdocs.go.list_user_docs.imports]
import (
	"os"

	"github.com/aws/aws-sdk-go/aws/session"
	"github.com/aws/aws-sdk-go/service/workdocs"

	"flag"
	"fmt"
)

// snippet-end:[workdocs.go.list_user_docs.imports]

func main() {
	// snippet-start:[workdocs.go.list_user_docs.vars]
	userPtr := flag.String("u", "", "User for whom info is retrieved")
	orgPtr := flag.String("o", "", "Your organization ID")

	flag.Parse()

	if *userPtr == "" || *orgPtr == "" {
		flag.PrintDefaults()
		os.Exit(1)
	}
	// snippet-end:[workdocs.go.list_user_docs.vars]

	// Initialize a session that the SDK will use to load
	// credentials from the shared credentials file. (~/.aws/credentials).
	// snippet-start:[workdocs.go.list_user_docs.session]
	sess := session.Must(session.NewSessionWithOptions(session.Options{
		SharedConfigState: session.SharedConfigEnable,
	}))

	svc := workdocs.New(sess)
	// snippet-end:[workdocs.go.list_user_docs.session]

	// snippet-start:[workdocs.go.list_user_docs.root_folder]
	input := new(workdocs.DescribeUsersInput)
	input.OrganizationId = orgPtr
	input.Query = userPtr

	result, err := svc.DescribeUsers(input)
	if err != nil {
		fmt.Println("Error getting user info", err)
		return
	}

	var folderID = ""

	if *result.TotalNumberOfUsers == 1 {
		for _, user := range result.Users {
			folderID = *user.RootFolderId
		}
		// snippet-end:[workdocs.go.list_user_docs.root_folder]

		// snippet-start:[workdocs.go.list_user_docs.describe]
		result, err := svc.DescribeFolderContents(&workdocs.DescribeFolderContentsInput{FolderId: &folderID})

		if err != nil {
			fmt.Println("Error getting docs for user", err)
			return
		}

		fmt.Println(*userPtr + " docs:")
		fmt.Println("")

		for _, doc := range result.Documents {
			fmt.Println(*doc.LatestVersionMetadata.Name)
			fmt.Println("  Size:         ", *doc.LatestVersionMetadata.Size, "(bytes)")
			fmt.Println("  Last modified:", *doc.LatestVersionMetadata.ModifiedTimestamp)
			fmt.Println("")
		}
		// snippet-end:[workdocs.go.list_user_docs.describe]
	}
}

// snippet-end:[workdocs.go.list_user_docs.complete]

File: ruby/example_code/workdocs/wd_list_user_docs.rb
# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
# SPDX - License - Identifier: Apache - 2.0

# Purpose
# This code example demonstrates how to list the documents for each of your Amazon WorkDocs users.

# snippet-start:[s3.wd_list_user_docs.rb]

require 'aws-sdk-workdocs'  # v2: require 'aws-sdk'

def get_user_folder(client, orgId, user_email)
  root_folder = ''

  resp = client.describe_users({
    organization_id: orgId,
  })

  # resp.users should have only one entry
  resp.users.each do |user|
    if user.email_address == user_email
      root_folder = user.root_folder_id
    end
  end

  return root_folder
end
# Replace us-west-2 with the AWS Region you're using for Amazon WorkDocs.
client = Aws::WorkDocs::Client.new(region: 'us-west-2')

# Set to the email address of a user
user_email = 'someone@somewhere'

# Set to the OrganizationId of your WorkDocs site.
orgId = 'd-123456789c'

user_folder = get_user_folder(client, orgId, user_email)

if user_folder == ''
  puts 'Could not get root folder for user with email address ' + user_email
  exit(1)
end

resp = client.describe_folder_contents({
  folder_id: user_folder, # required
  sort: "NAME", # accepts DATE, NAME
  order: "ASCENDING", # accepts ASCENDING, DESCENDING
})

resp.documents.each do |doc|
  md = doc.latest_version_metadata

  puts "Name:          #{md.name}"
  puts "Size (bytes):  #{md.size}"
  puts "Last modified: #{doc.modified_timestamp}"
  puts "Doc ID:        #{doc.id}"
  puts "Version ID:    #{md.id}"
  puts
end
# snippet-end:[s3.wd_list_user_docs.rb]
